src/common/transformations/src/transformations/common_optimizations/sequence_fusion.cpp

```diff
@@ -0,0 +1,325 @@
+                                           first_cell->get_hidden_size(),
+                                           ov::op::RecurrentSequenceDirection::FORWARD,
+                                           first_cell->get_activations(),
+                                           first_cell->get_activations_alpha(),
+                                           first_cell->get_activations_beta(),
+                                           first_cell->get_clip());
+    } else if (dynamic_pointer_cast<ov::op::internal::AUGRUCell>(first_cell)) {
+        const auto A_in = cp_to.make<Concat>(attention_to_concat, 1);
+        sequence = cp_to.make<ov::op::internal::AUGRUSequence>(X_in,
+                                                               Ht_in,
+                                                               sequence_lengths_in,
+                                                               W_in,
+                                                               R_in,
+                                                               B_in,
+                                                               A_in,
+                                                               first_cell->get_hidden_size());
+    } else {
+        // cell is not supported;
+        return false;
+    }
+
+    outputs[0] = cp_to.make<Squeeze>(sequence->output(1), axis_1);
+    replace_outputs_update_names(last_cell->outputs(), outputs);
+
+    if (!h_inputs_to_redirect.empty()) {
+        auto squeeze_Y = cp_to.make<Squeeze>(sequence->output(0), axis_1);
+        auto split = cp_to.make<Split>(squeeze_Y, axis_1, cells_cnt);
+
+        for (const auto& it : h_inputs_to_redirect) {
+            for (const auto& in : it.second) {
+                auto squeeze = cp_to.make<Squeeze>(split->output(cells_cnt - it.first - 1), axis_1);
+                in.replace_source_output(squeeze);
+            }
+        }
+    }
+    return true;
+}
+}  // namespace
+
+ov::pass::SequenceFusion::SequenceFusion() {
+    MATCHER_SCOPE(SequenceFusion);
+
+    auto cell = wrap_type<RNNCellBase>();
+    ov::matcher_pass_callback callback = [=](pattern::Matcher& m) {
+        NodeRegister copy_from;
+        NodeRegister copy_to;
+        auto cell = m.get_match_root();
+        shared_ptr<RNNCellBase> current_cell = dynamic_pointer_cast<RNNCellBase>(cell);
+        if (!current_cell) {
+            return false;
+        }
+
+        // check that this is the last Cell in the chain, e.g.
+        // GRUCell -> GRUCell (the last cell) -> OtherNode
+        // GRUCell (hidden_size = 128) -> GRUCell (hs = 128, the last) -> GRUCell (hs = 64)
+        for (const auto& target : cell->get_output_target_inputs(0)) {
+            auto cell_1 = dynamic_pointer_cast<RNNCellBase>(target.get_node()->shared_from_this());
+            if (cell_1 && is_equal_cells(cell_1, current_cell)) {
+                return false;
+            }
+        }
+
+        int cells_cnt = 0;
```

Comment:
I think change it to 
```suggestion
        int cells_cnt;
```
As it should be set in the function below. If not happens e.g. static analysis tools should complain that working on uninitialized variable which can be suggestion of programming error set it will mask all issues.